baselines/flwr_baselines/publications/wav2vec2/tedlium/model/model.py

```diff
@@ -0,0 +1,132 @@
+"""Provides function for using a SpeechBrain model."""
+
+import gc
+import os
+
+import speechbrain as sb
+import torch
+from flwr.common import ndarrays_to_parameters
+from hyperpyyaml import load_hyperpyyaml
+
+from model.sb_w2v2 import ASR, get_weights
+from preprocessing.data_loader import (
+    dataio_prepare,
+)
+
+
+def int_model(  # pylint: disable=too-many-arguments,too-many-locals
+    cid,
+    device,
+    config_path,
+    save_path,
+    data_path,
+    label_path=None,
+    parallel=True,
+    evaluate=False,
+):
+    """Setting up the experiment.
+
+    Loading the hyperparameters from config files and
+    command-line overrides, setting the correct path for the
+    corresponding clients, and creating the model.
+    """
+    # Load hyperparameters file with command-line overrides
+    save_path = save_path + "client_" + str(cid)
+    # Override with FLOWER PARAMS
+    if evaluate:
+        overrides = {
+            "output_folder": save_path,
+            "number_of_epochs": 1,
+            "test_batch_size": 4,
+            "device": device,
+        }
+
+    else:
+        overrides = {"output_folder": save_path}
+
+    if label_path is None:
+        label_path = os.path.join(save_path, "label_encoder.txt")
+
+    _, run_opts, _ = sb.parse_arguments(config_path)
+    run_opts["device"] = device
+    run_opts["data_parallel_backend"] = parallel
+    run_opts["noprogressbar"] = True # disable tqdm progress bar
+
+    with open(config_path) as fin:
+        params = load_hyperpyyaml(fin, overrides)
+
+    # This logic follow the data_path is a path to csv folder file
+    # All train/dev/test csv files are in the same name format for server and client
+    # Example:
+    # server: /users/server/train.csv
+    # client: /users/client_1/train.csv
+    # Modify (if needed) the if else logic to fit with path format
+
+    if int(cid) != 19999:
+        params["data_folder"] = os.path.join(data_path, "client_" + str(cid))
+    else:
+        params["data_folder"] = os.path.join(data_path, "client_" + str(1300))
+
+    params["train_csv"] = params["data_folder"] + "/ted_test.csv"
+    params["valid_csv"] = params["data_folder"] + "/ted_test.csv"
+    params["test_csv"] = params["data_folder"] + "/ted_test.csv"
```

Comment:
```suggestion
    else:
        params["data_folder"] = os.path.join(data_path, "server")

    params["valid_csv"] = params["data_folder"] + "/ted_test.csv"
    params["test_csv"] = params["data_folder"] + "/ted_test.csv"

    if int(cid) < 1341:
        params["train_csv"] = params["data_folder"] + "/ted_train_wo5.csv"
    else:
        params["train_csv"] = params["data_folder"] + "/ted_train.csv"
    
```

To re-produce exact result